src/CentralNic/Provider.php

```diff
@@ -151,21 +153,25 @@ public function register(RegisterDomainParams $params): DomainResult
 
             return $this->_getInfo($domainName, sprintf('Domain %s was registered successfully!', $domainName));
         } catch (eppException $e) {
-            return $this->_eppExceptionHandler($e, $params->toArray());
+            $this->_eppExceptionHandler($e, $params->toArray());
         }
     }
 
+    /**
+     * @throws \Upmind\ProvisionBase\Exception\ProvisionFunctionError
+     */
     private function getRegisterParams(RegisterDomainParams $params): array
     {
         if (Arr::has($params, 'registrant.id')) {
             $registrantID = $params->registrant->id;
 
-            if (!$this->api()->getContactInfo($registrantID)) {
```

Comment:
@uphlewis the `!` check didn't make sense, so checking whether data are returned